src/modules/workspace/promote.ts

```diff
@@ -39,11 +38,12 @@ const promptPromoteConfirm = (workspace: string): Promise<any> =>
     }
   )
 
-export default () => {
+export default async () => {
   log.debug('Promoting workspace', currentWorkspace)
-  return isPromotable(currentWorkspace)
+  await isPromotable(currentWorkspace)
     .then(() => promptPromoteConfirm(currentWorkspace))
     .then(() => promote(account, currentWorkspace))
```

Comment:
Why do you need to await? For me it's odd to use await and also `.then` when you can either:
```
await isPromotable(currentWorkspace)
promptPromoteConfirm(currentWorkspace)
promote(account, currentWorkspace)
```
or
```
return isPromotabel(currentWorkspace).then(...)
```